### DISCUSSION TITLE
Activation event doesn't fire *second* time user clicks extension icon in activitybar

### DISCUSSION THREAD

**Discussion by jasongonzales23** [microsoft/vscode-discussions#52] [Extension Development QnA] [answered]
I need some help understanding what I need to do to get my activation event to fire when a user clicks on the activitybar icon for my extension. Not even "*" seems to make it run.

When I click the icon, my sidebar appears, but no activation event runs.

I suspect I'm doing something wrong but not sure where to start. Should I be doing something with deactivate?

**Comment by jasongonzales23** [comment]
Video to demonstrate. Something, something, a thousand words....

https://user-images.githubusercontent.com/243861/176975743-2e0c2cab-73e5-45e0-bcc0-1a6bafe0bc90.mp4

**Comment by isidorn** [comment]
Are you still facing this issue?
These docs might help https://code.visualstudio.com/api/references/activation-events

If you are still facing the issue, why do you think your extension is not activated?

**Comment by jasongonzales23** [comment]
It's still happening as shown in the video above. I haven't had time to dig into it, but my sense is that the issue I'm showing above is less about the activation event and more about how I'm keeping track of that WebviewPanel. Because VSC is so object oriented and I'm more of a straight up JS and more functional programmer I may not be keeping track of that instance properly perhaps. For the record I absolutely hate object oriented programming so I tend to write it with anger and disgust in my heart and it therefore shows in my code. Apologies to all lovers of OOP.

**Reply by olguzzar** [reply]
Hi there, following up on this as it's been a while since this post has had activity. Is this still an issue?

**Comment by jasongonzales23** [comment]
Yes, it's still an issue in the most recent version. Thanks.

Jason Gonzales
ph: 415.858.0207

**Comment by isidorn** [comment, marked as answer]
Why do you want 2 activation events here? I suggest to file a new issue in the VS Code repo with a minimalistic reporoducable case.
